### PR TITLE
Github action deployment for hack-zero backend

### PR DESCRIPTION
**Pull request by HuangStanley050**
trying to see if github action picks up the custom variables depending on the stage set for deploy

only trying for two variables MONGO_USER and MONGO_PASSWORD

reference from : https://adamdelong.com/serverless-environment-variables/

# Serverless Environment Variables Based on Serverless Stage

### CODE REVIEW

**Review by jpeyper — APPROVED** [review]

**Comment by jpeyper** [comment]
you can setup github actions to run to on PR rather than master merges if you want test stuff without merging

**Comment by HuangStanley050** [comment]
> you can setup github actions to run to on PR rather than master merges if you want test stuff without merging

great idea! I will try that.

**Comment by HuangStanley050** [comment]
didn't work, back to the drawing board but good thing we didn't merge this to master

**Comment by HuangStanley050** [comment]
some improvement, could at least get github action serverless deploy to read the .env.production file however still fails, not able to login to aws console from my personal computer, will need to use work computer tomorrow and check the aws console on env variables for the lambda

image

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -2,18 +2,27 @@ on:
...
+
       - uses: actions/checkout@master
       - run: npm install
+
       - name: serverless deploy
+
         uses: serverless/github-action@master
         with:
-          args: deploy
+          args: NODE_ENV=production deploy -s production
```
Comment (jpeyper) [inline .github/workflows/main.yml]: set the env variable in the env section below, not here.

**Comment by HuangStanley050** [comment]
confirm the decryption ran and the file is sitting in the directory but for some reason sls is still not able pick it up despite setting the NODE_ENV to production

image

**Comment by HuangStanley050** [comment]
looks like the file created in the previous step before serverless deploy is not carried over for some reason....

**Comment by HuangStanley050** [comment]
ok, the problem is how to get serverless's config to read from the github action's file system, the .env.production is stored in $HOME/secrets but I am still not able to read that during serverless deployment.

**Comment by HuangStanley050** [comment]
need to figure out how to read the data from $HOME/secrets/.env.production

**Comment by HuangStanley050** [comment]
confirm file sitting in $HOME/secrets just need to somehow get serverless.yml to read from there!!

image

**Review by danieltongaus — COMMENTED** [review]
LGTM

File: serverless.yml [hunk 2]
```diff
@@ -23,39 +15,22 @@ provider:
...
-#       Ref: CognitoUserPool
+
+environment:
+  MONGO_USER: ${env:MONGO_USER}
+  MONGO_PASSWORD: ${env:MONGO_PASSWORD}
+  AWS_ACCESS_KEY_ID: ${env:AWS_ACC_KEY_ID}
+  AWS_SECRET: ${env:AWS_SECRET}
+  AWS_USER_POOL_ID: ${env:AWS_USER_POOL_ID}
+  ACCESS_CONTROL_ALLOW_ORIGIN: ${env:ACCESS_CONTROL_ALLOW_ORIGIN}
+  HOME: ${evn:HOME}
```
Comment (jpeyper) [inline serverless.yml]: shouldn't this be `${env:HOME}`

Comment (HuangStanley050) [reply serverless.yml]: Fixed typo but something sus about how we pass in the github $HOME to the main.yml, have reached out to DT to help but agreed at standup that if it needs couple of hours to resolve, we will drop it for now.

**Comment by HuangStanley050** [comment]
It's a success, the issue was I used a checkout action prior to sls deploy that reset the workspace which decrypt action was ran. Now I need to work out how to conditionally deploy dev and prod environment

**Comment by HuangStanley050** [comment]
confirm working, developing locally sls will pick up the .env file but when deploy via github action it will read from .env.production which is produced by github actions.

**Pull request merged**